src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs

```diff
@@ -107,10 +108,15 @@ private protected override void ChangeVisualState(bool useTransitions)
 				isOn = IsOn;
 				GoToState(useTransitions, isOn ? "On" : "Off");
 				GoToState(useTransitions, isOn ? "OnContent" : "OffContent");
+
 				// TODO Uno specific: We must force the knob to position, because
 				// we don't yet support the transitions which do this automatically
 				// in XAML template.
-				ForceSwitchKnobEndPosition();
+				if (IsLoaded)
+				{
+					Debug.Assert(GetMaxOffset() > 0);
+					ForceSwitchKnobEndPosition();
+				}
```

Comment:
Are we able to validate this somehow by inspecting the children positions?

Reply:
Can we also add tests for this @Youssef1313 / @jeromelaban ?

Reply:
I'll try with getting `_minKnobTranslation` and `_maxKnobTranslation` via reflection and assert against these. This is the core thing that breaks ToggleSwitch. The intent of this change isn't only about the "initial" position of the ToggleSwitch, but it's also when dragging, we get negative _minKnobTranslation and then dragging is only allowed in the reverse direction (outside the bounds of the control)